parlai/mturk/webapp/run_mocks/mock_turk_agent.py

```diff
@@ -0,0 +1,311 @@
+            AssignState.STATUS_DONE,
+            AssignState.STATUS_PARTNER_DISCONNECT
+        ]
+
+    def is_final(self):
+        """Determine if this agent is in a final state"""
+        return self.state.is_final()
+
+    def append_message(self, message):
+        """Add a received message to the state"""
+        self.state.append_message(message)
+
+    def set_last_command(self, command):
+        """Changes the last command recorded as sent to the agent"""
+        self.state.set_last_command(command)
+
+    def get_last_command(self):
+        """Returns the last command to be sent to this agent"""
+        return self.state.get_last_command()
+
+    def clear_messages(self):
+        """Clears the message history for this agent"""
+        self.state.clear_messages()
+
+    def get_messages(self):
+        """Returns all the messages stored in the state"""
+        return self.state.get_messages()
+
+    def get_connection_id(self):
+        """Returns an appropriate connection_id for this agent"""
+        return "{}_{}".format(self.worker_id, self.assignment_id)
+
+    def log_reconnect(self):
+        pass
+
+    def get_inactive_command_data(self):
+        """Get appropriate inactive command data to respond to a reconnect"""
+        text, command = self.state.get_inactive_command_text()
+        return {
+            'text': command,
+            'inactive_text': text,
+            'conversation_id': self.conversation_id,
+            'agent_id': self.worker_id,
+        }
+
+    def wait_for_status(self, desired_status):
+        """Suspend a thread until a particular assignment state changes
+        to the desired state
+        """
+        while True:
+            if self.get_status() == desired_status:
+                return True
+            if self.is_final():
+                return False
+            time.sleep(shared_utils.THREAD_SHORT_SLEEP)
+
+    def is_in_task(self):
+        return self.status == AssignState.STATUS_IN_TASK
+
+    def observe(self, msg):
+        """Send an agent a message through the mturk_manager"""
+        self.mturk_manager.send_message(
+            self.worker_id, self.assignment_id, msg)
+
+    def put_data(self, id, data):
+        """Put data into the message queue if it hasn't already been seen"""
+        self.msg_queue.put(data)
+
+    def flush_msg_queue(self):
+        """Clear all messages in the message queue"""
+        if self.msg_queue is None:
+            return
+        while not self.msg_queue.empty():
+            self.msg_queue.get()
+
+    def reduce_state(self):
+        """Cleans up resources related to maintaining complete state"""
+        self.flush_msg_queue()
+        self.msg_queue = None
+
+    def get_new_act_message(self):
+        """Get a new act message if one exists, return None otherwise"""
+        # See if any agent has disconnected
+        if self.disconnected or self.some_agent_disconnected:
+            msg = {
```

Comment:
this format is used a lot, you could just have some constant "MSG" and edit the text, returning as needed

Reply:
I'm going to subclass the real MTurkAgent and standardize this because it can be cleaner for sure.